crypto/src/signature.rs

```diff
@@ -82,113 +87,91 @@ impl Signature {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        self.signature.partial_cmp(&other.signature)
+impl<T> Clone for SignatureOf<T> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone(), PhantomData)
     }
 }
-impl<T> Ord for SignatureOf<T> {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        self.signature.cmp(&other.signature)
+
+impl<T> PartialEq for SignatureOf<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.eq(&other.0)
     }
 }
+impl<T> Eq for SignatureOf<T> {}
 
-impl<T> Encode for SignatureOf<T> {
-    fn size_hint(&self) -> usize {
-        Signature::size_hint(&self.0)
-    }
-    fn encode_to<U: parity_scale_codec::Output + ?Sized>(&self, dest: &mut U) {
-        self.0.encode_to(dest)
-    }
-    fn encode(&self) -> Vec<u8> {
-        self.0.encode()
-    }
-    fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
-        self.0.using_encoded(f)
+impl<T> PartialOrd for SignatureOf<T> {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        self.0.partial_cmp(&other.0)
     }
-    fn encoded_size(&self) -> usize {
-        self.0.encoded_size()
+}
+impl<T> Ord for SignatureOf<T> {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0.cmp(&other.0)
     }
 }
 
-impl<T> Decode for SignatureOf<T> {
-    fn decode<I: parity_scale_codec::Input>(
-        input: &mut I,
-    ) -> Result<Self, parity_scale_codec::Error> {
-        Ok(Self(Signature::decode(input)?, PhantomData))
-    }
-    fn skip<I: parity_scale_codec::Input>(input: &mut I) -> Result<(), parity_scale_codec::Error> {
-        Signature::skip(input)
-    }
-    fn encoded_fixed_size() -> Option<usize> {
-        Signature::encoded_fixed_size()
+impl<T> IntoSchema for SignatureOf<T> {
+    fn schema(metamap: &mut iroha_schema::MetaMap) {
+        Signature::schema(metamap)
     }
 }
 
 impl<T> SignatureOf<T> {
-    /// Verifies signature for this hash
+    /// Creates new [`SignatureOf`] by signing value via [`KeyPair::private_key`]
+    ///
     /// # Errors
-    /// Fails if verification fails
-    pub fn verify_hash(&self, hash: &HashOf<T>) -> Result<(), SignatureVerificationFail<T>> {
-        self.0
-            .verify(hash.as_ref())
-            .map_err(|err| SignatureVerificationFail {
-                signature: Box::new(self.clone()),
-                reason: err.to_string(),
-            })
+    /// Fails if decoding digest of key pair fails
+    #[cfg(feature = "std")]
+    pub fn from_hash(key_pair: KeyPair, hash: &HashOf<T>) -> Result<Self, Error> {
+        Ok(Self(Signature::new(key_pair, hash.as_ref())?, PhantomData))
     }
 
     /// Transmutes signature to some specific type
-    /// SAFETY:
-    /// Do at your own risk
     pub fn transmute<F>(self) -> SignatureOf<F> {
         SignatureOf(self.0, PhantomData)
     }
 
     /// Transmutes signature to some specific type
+    ///
     /// SAFETY:
```

Comment:
```suggestion
    /// # Safety
```

Reply:
This is the only place in our codebase we use `unsafe` as of today. We should be careful here

I would even like to remove the warning from these transmutes. Since I've added `repr(transparent)` the ABI is guaranteed to match for all transmuted types so this code should be safe to compile. The only possible issue is if types need to uphold some invariants but that's not our concern anyways

Reply:
Maybe just replace it with 
```
/// # Warning: unsafe
```
And be done with it?